### ISSUE TITLE
ISL on VS Code Crashes 

### ISSUE THREAD

**Issue by zengk95** [facebook/sapling#477]
Sometimes the ISL crashes randomly (or not in any pattern I can recall). Once it happens, the page will become inconsistent with the web version and the sl status command. If I try to do anything on it, nothing will really happen.

I'm using 
Sapling 0.1.20221201-095354-r360873f1
Windows 10
VS Code version 1.74.2
e8a3071ea4344d9d48ef8a4df2c097372b0c5161
x64

Error Logs in output
```
conflict state still the same (NOT in conflict)
fetched 5 github PR summaries
Error fetching files:  [Error: Webview is disposed
	at i.c (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:96:170431)
	at get webview [as webview] (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:96:169402)
	at Object.postMessage (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:104307)
	at Si.postMessage (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:98337)
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:99856
	at vn.t (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:84837)
	at vn.emit (node:events:538:35)
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:78625
	at i (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:41621)
	at Fn.fetchUncommittedChanges (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:41714)
	at h.changeCallback (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:80536)
	at h.poll (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:47475)
	at t.setState (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:41262)
	at Fn.setPageFocus (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:83880)
	at Si.handleIncomingMessageWithRepo (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:101571)
	at Si.handleIncomingMessage (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:98883)
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:97799
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:104388
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at r.$onMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:96:161394)
	at u.N (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:11208)
	at u.M (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:10926)
	at u.H (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:10019)
	at u.G (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:9157)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:7788
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at h.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:14314)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:120:15804
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at h.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:14314)
	at c.y (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:17324)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:15795
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at g.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:12045)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:11332
	at Socket.l (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:19792)
	at Socket.emit (node:events:526:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Readable.push (node:internal/streams/readable:228:10)
	at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)]
Error fetching commits:  [Error: Webview is disposed
	at i.c (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:96:170431)
	at get webview [as webview] (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:96:169402)
	at Object.postMessage (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:104307)
	at Si.postMessage (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:98337)
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:100321
	at vn.t (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:84639)
	at vn.emit (node:events:538:35)
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:78995
	at i (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:41621)
	at Fn.fetchSmartlogCommits (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:41714)
	at h.changeCallback (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:80567)
	at h.poll (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:47475)
	at t.setState (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:41262)
	at Fn.setPageFocus (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:83880)
	at Si.handleIncomingMessageWithRepo (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:101571)
	at Si.handleIncomingMessage (c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:98883)
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:97799
	at c:\Users\kzeng\.vscode\extensions\meta.sapling-scm-0.1.8\dist\extension.js:1:104388
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at r.$onMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:96:161394)
	at u.N (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:11208)
	at u.M (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:10926)
	at u.H (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:10019)
	at u.G (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:9157)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:104:7788
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at h.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:14314)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:120:15804
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at h.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:14314)
	at c.y (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:17324)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:15795
	at d.invoke (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at w.deliver (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2029)
	at g.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1667)
	at g.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:12045)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:11332
	at Socket.l (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:72:19792)
	at Socket.emit (node:events:526:28)
	at addChunk (node:internal/streams/readable:315:12)
	at readableAddChunk (node:internal/streams/readable:289:9)
	at Readable.push (node:internal/streams/readable:228:10)
	at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)]
checking for merge conflicts
fetching github PR summaries
conflict state still the same (NOT in conflict)
fetched 5 github PR summaries
```

**Comment by bolinfest** [comment]
@zengk95 FYI, 0.1.20221201-095354-r360873f1 is from Dec 1 while the version from Dec 22 includes a number of fixes: https://github.com/facebook/sapling/releases/tag/0.2.20221222-152408-ha6a66d09.

**Comment by zengk95** [comment]
I've updated it to Sapling 0.2.20221222-152408-ha6a66d09. 
![image](https://user-images.githubusercontent.com/34172846/213524650-6598d8a4-9073-4bc9-b139-0fad30670270.png)

It's still giving me some issues. This Error webview is disposed happens basically anytime I click on any button like refresh or goto a commit but doesn't actually do anything. 


Also another weird thing is I could've sworn we had a way to revert a file in the ISL while hovering over a file on the sidebar  (at least on the vs code extension) but I don't see the option to anymore.

**Comment by bolinfest** [comment]
To clarify, do you have problems when using `sl web` and using ISL in the browser or only when using it in VS Code? The "Webview is disposed" error should be specific to VS Code, I believe?

**Comment by zengk95** [comment]
sl web works fine. It's only on VS Code.

**Comment by evangrayk** [comment]
I fixed the errors about `Webview is disposed` in 3102532504e34f4c06561005446e3736f91d2c43, but there's likely still some unreliability issue that's affecting you. I'll keep looking into it.

**Comment by evangrayk** [comment]
@zengk95 I've been testing the vscode extension for a couple of weeks after 3102532504e34f4c06561005446e3736f91d2c43, and it seems like this issue no longer reproduces for me. I just published VS Code extension version 0.1.9 with this fix (and others). Let me know if this helps fix this issue!

**Comment by zengk95** [comment]
I've been using it for a bit and it seems to work pretty well now! Thanks for addressing this!
